plugins/woocommerce/src/StoreApi/Utilities/AgenticCheckoutUtils.php

```diff
@@ -0,0 +1,342 @@
+		$customer->set_shipping_city( $city );
+		$customer->set_shipping_state( $state );
+		$customer->set_shipping_postcode( $postal_code );
+		$customer->set_shipping_country( $country );
+
+		// Also set as billing address if not already set.
+		if ( ! $customer->get_billing_address_1() ) {
+			// For billing, only set names if provided or use existing billing names.
+			if ( ! empty( $address['name'] ) ) {
+				$customer->set_billing_first_name( $first_name );
+				$customer->set_billing_last_name( $last_name );
+			}
+			$customer->set_billing_address_1( $line_one );
+			$customer->set_billing_address_2( $line_two );
+			$customer->set_billing_city( $city );
+			$customer->set_billing_state( $state );
+			$customer->set_billing_postcode( $postal_code );
+			$customer->set_billing_country( $country );
+		}
+
+		$customer->save();
+	}
+
+	/**
+	 * Clear fulfillment address from customer.
+	 *
+	 * @param \WC_Customer $customer Customer instance.
+	 */
+	public static function clear_fulfillment_address( $customer ) {
+		// Clear shipping address.
+		$customer->set_shipping_first_name( '' );
+		$customer->set_shipping_last_name( '' );
+		$customer->set_shipping_address_1( '' );
+		$customer->set_shipping_address_2( '' );
+		$customer->set_shipping_city( '' );
+		$customer->set_shipping_state( '' );
+		$customer->set_shipping_postcode( '' );
+		$customer->set_shipping_country( '' );
+
+		$customer->save();
+	}
+
+	/**
+	 * Add Agentic Commerce Protocol headers to response.
+	 *
+	 * @param \WP_REST_Response $response Response object.
+	 * @param \WP_REST_Request  $request Request object.
+	 * @return \WP_REST_Response Response with headers.
+	 */
+	public static function add_protocol_headers( \WP_REST_Response $response, \WP_REST_Request $request ) {
+		// Echo Idempotency-Key header if provided.
+		$idempotency_key = $request->get_header( 'Idempotency-Key' );
+		if ( $idempotency_key ) {
+			$response->header( 'Idempotency-Key', $idempotency_key );
+		}
+
+		// Echo Request-Id header if provided.
+		$request_id = $request->get_header( 'Request-Id' );
+		if ( $request_id ) {
+			$response->header( 'Request-Id', $request_id );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Check if the Agentic Checkout feature is enabled.
+	 *
+	 * V1 implementation: Returns true if feature is enabled (no auth check).
+	 * Future: Implement Bearer token authentication.
+	 *
+	 * @return bool|\WP_Error True if authorized, WP_Error otherwise.
+	 */
+	public static function is_authorized() {
+		// Check if feature is enabled.
+		$features_controller = wc_get_container()->get( FeaturesController::class );
+		if ( ! $features_controller->feature_is_enabled( 'agentic_checkout' ) ) {
+			return new \WP_Error(
+				'woocommerce_rest_agentic_checkout_disabled',
+				__( 'Agentic Checkout API is not enabled.', 'woocommerce' ),
+				array( 'status' => 403 )
+			);
+		}
+
+		// V1: Allow all requests (implement proper auth in future).
+		return true;
```

Comment:
Can this route be abused in anyway? Thinking about exposing data, unauthorized usage, spam etc..

Reply:
This is a todo as @ebinnion commented above https://github.com/woocommerce/woocommerce/pull/61216#discussion_r2400571539, and in the PR description, this will be addressed.